File: Arrays_And_Strings/Running_Sum_Of_1d_Array/Solution.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> runningSum(std::vector<int>& nums) {
        std::vector<int> sum{nums[0]};
        for(int i = 1; i < nums.size(); i++) {
            //sum[i] = sum[i-1] + nums[i];
            sum.push_back(sum[i-1] + nums[i]);
        }            
        return sum;
    }
};

int main() {
    std::vector<int> nums {3,1,2,10,1};
    Solution s1;
    std::vector<int> result;
    result = s1.runningSum(nums);
    for(auto x: result) {
        std::cout << x << std::endl;
    }
    return 0;
}

File: Arrays_And_Strings/Running_Sum_Of_1d_Array/py/solution.py
from typing import List

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        for i in range(1, len(nums)):
            nums[i] = nums[i-1] + nums[i]
        return nums

if __name__ == "__main__":
    print("hello")
    s1 = Solution()
    nums = [1,2,3,4]
    ans = s1.runningSum(nums);
    for i in range(len(nums)):
        print(nums[i])
